firmware-support/bittide-hal-c/c_src/bittide_uart.c

```diff
@@ -75,3 +75,18 @@ void uart_putdec(Uart uart, uint64_t val) {
 
   uart_puts(uart, &buf[i]);
 }
+
+void uart_putdec_signed(Uart uart, int64_t val) {
+  if (val < 0) {
+    uart_putc(uart, '-');
+    // Handle most negative value specially to avoid overflow
+    // INT64_MIN is -9223372036854775808, which can't be negated in int64_t
+    if (val == (int64_t)0x8000000000000000ULL) {
+      uart_putdec(uart, 9223372036854775808ULL);
```

Comment:
How does this work? `INT64_MAX` is `9223372036854775807LL` so the literal can't be represented as anything.